### PR TITLE
Simplify sample_test.yaml

### PR DESCRIPTION
**Pull request by numerology**
Refactor sample_test.yaml using withItems syntax. The functions should not be impacted. This is part of the efforts of KFP sample tests.

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/pipelines/1709)

### CODE REVIEW

File: test/sample_test.yaml [hunk 1]
```diff
@@ -19,184 +19,120 @@ metadata:
...
+                  value: "{{inputs.parameters.namespace}}"
+                - name: sample-tests-image
+                  value: "{{inputs.parameters.target-image-prefix}}{{inputs.parameters.sample-tests-image-suffix}}"
+                - name: target-image-prefix
+                  value: "{{inputs.parameters.target-image-prefix}}"
+                - name: test-results-gcs-dir
+                  value: "{{inputs.parameters.test-results-gcs-dir}}"
+                - name: test-name
+                  value: "{{item.test-name}}"
+            withItems:
```
Comment (gaoning777) [inline test/sample_test.yaml]: Is there particular reason to use a list of dict? 
How about the following
withItems:
  - "tf-training"
  - "tfx"
If we need multiple params besides the test name in the future, we can convert it to a dict.
WDYT?

Comment (numerology) [reply test/sample_test.yaml]: Totally agree with you that currently we can do a list of str given the simple structure of params we have.
I chose dict cause I personally found the name 'item' is not very meaningful. PTAL.

**Comment by numerology** [comment]
Thanks for the review!

**Pull request merged**